validator/client/src/test/java/tech/pegasys/teku/validator/client/restapi/apis/GetGasLimitTest.java

```diff
@@ -0,0 +1,117 @@
+import static tech.pegasys.teku.validator.client.restapi.ValidatorTypes.PARAM_PUBKEY_TYPE;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.infrastructure.json.JsonUtil;
+import tech.pegasys.teku.infrastructure.restapi.StubRestApiRequest;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.validator.client.BeaconProposerPreparer;
+
+public class GetGasLimitTest {
+
+  private final Spec spec = TestSpecFactory.createMinimalBellatrix();
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+
+  private final BeaconProposerPreparer beaconProposerPreparer = mock(BeaconProposerPreparer.class);
+
+  private final GetGasLimit handler = new GetGasLimit(Optional.of(beaconProposerPreparer));
+  private final BLSPublicKey publicKey = dataStructureUtil.randomPublicKey();
+  private final StubRestApiRequest request =
+      StubRestApiRequest.builder()
+          .metadata(handler.getMetadata())
+          .pathParameter("pubkey", publicKey.toBytesCompressed().toHexString())
+          .build();
+
+  private final UInt64 gasLimit = dataStructureUtil.randomUInt64();
+
+  @Test
+  void shouldEncodePubkey() throws JsonProcessingException {
+    final String keyString = JsonUtil.serialize(publicKey, PARAM_PUBKEY_TYPE.getType());
+    assertThat(keyString).isEqualTo("\"" + publicKey.toBytesCompressed().toHexString() + "\"");
+  }
+
+  @Test
+  void shouldDecodePubkey() {
+    final BLSPublicKey pubkey = request.getPathParameter(PARAM_PUBKEY_TYPE);
+    assertThat(pubkey).isEqualTo(publicKey);
+  }
+
+  @Test
+  void metadata_shouldHandle400() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_BAD_REQUEST);
+  }
+
+  @Test
+  void metadata_shouldHandle500() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  void metadata_shouldHandle404() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_NOT_FOUND);
+  }
+
+  @Test
+  void shouldRespondOk() throws JsonProcessingException {
+    when(this.beaconProposerPreparer.getGasLimit(any())).thenReturn(Optional.of(gasLimit));
+    handler.handleRequest(request);
+    assertThat(request.getResponseCode()).isEqualTo(SC_OK);
+    assertThat(request.getResponseBody())
+        .isEqualTo(new GetGasLimit.GetGasLimitResponse(gasLimit, publicKey));
+  }
+
+  @Test
+  void metadata_shouldHandle200() throws JsonProcessingException {
+    final String responseData =
+        getResponseStringFromMetadata(
+            handler,
+            SC_OK,
+            new GetGasLimit.GetGasLimitResponse(gasLimit, dataStructureUtil.randomPublicKey()));
```

Comment:
reuse the publickey we have already in the test, and then can reference it in the test like you reference gaslimit...